### PR TITLE
convert auth action file to TS and add tests

### PR DESCRIPTION
**Pull request by EllieLilithFox**

### CODE REVIEW

File: gearcatalog/frontend/src/actions/gear-list.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import axios from "axios";
 import { createMessage, returnErrors } from "./messages";
 import { tokenConfig } from "./auth";
 
-import * as c from "./types";
+import * as c from "./action-constants";
```
Comment (NatalieWolfe) [inline gearcatalog/frontend/src/actions/gear-list.js:5]: Why did you rename this file?

**Review by NatalieWolfe — APPROVED** [review]

File: gearcatalog/frontend/src/actions/gear-list.js (hunk 1 above)
Comment (EllieLilithFox) [reply gearcatalog/frontend/src/actions/gear-list.js:5]: There's bleed over with having a TS file named `action.types` I changed it to clear up any confusion. They are constants for action types so felt that should be more clear than having two files with types in their name.
